### PR TITLE
Fix completion API based on API review.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/12746

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Features/Core/Portable/Completion/CompletionChange.cs [hunk 1]
```diff
@@ -41,9 +41,6 @@ private CompletionChange(ImmutableArray<TextChange> textChanges, int? newPositio
         private CompletionChange(TextChange textChange, int? newPosition, bool includesCommitCharacter)
         {
             TextChange = textChange;
-#pragma warning disable CS0618 // Type or member is obsolete
-            TextChanges = ImmutableArray.Create(textChange);
```
Comment (Pilchie) [inline src/Features/Core/Portable/Completion/CompletionChange.cs]: Not setting this break binary compat (behavior-wise), for people who haven't recompiled yet.

**Comment by Pilchie** [comment]
What about the point in #12746 about passing the position in?

File: src/Features/Core/Portable/Completion/CompletionChange.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Completion/CompletionChange.cs]: Our feeling is that this API simply isn't one htat people will really be making use of.  It's too new, and it's still too difficult to be a roslyn service language *provider* (as opposed to consumer).

**Comment by CyrusNajmabadi** [comment]
@Pilchie We discussed it with @jaredpar .  We're ok with not passing the positoin in.

**Pull request merged**